Wox.Infrastructure/StringMatcher.cs

```diff
@@ -164,12 +151,42 @@ public static int ScoreForPinyin(string source, string target)
     public class MatchResult
     {
         public bool Success { get; set; }
-        public int Score { get; set; }
+
+        private int _score;
+        public int Score
+        {
+            get
+            {
+                return _score;
+            }
+            set
+            {
+                _score = ApplySearchPrecisionFilter(value);
```

Comment:
I dont think it is a good approach to apply the filter here, i get that this call is used pretty much every time, but applying it here hidden in the setter makes it harder to reason about what score is, if i set 10 i expect the value to be 10 not possible 10 or 0. Wouldn't it be better to know score is 10 and after i apply the precision filter then it could be the filtered score? 

I always taken that Score should behave as a raw score after score calculation then we can enhance the MatchResult if we want but be more explicit about the change/filtering. What do you think

Reply:
I see your point here.
I was basically trying to remove the need to call "ScoreAfterSearchPrecisionFilter()" every time a filtered score is needed. This method call is something we cannot guarantee for every plugin there is, but as the search precision setting is always active, we need to take care of that in some way beforehand.

We could add another property called "RawScore" for example which is used to store the value before applying the search precision filter. What do you think?

Reply:
Yes 'RawScore' is what I was thinking, let's go with that for now. What do you think of one of the below options:
 1. Mark Score obsolete, add 'RawScore' and add 'AdjustedScore'.
Or
2. Keep Score as long as it is clear that it is derived from applying precision filter from RawScore.

Reply:
I think both options are fine and would make it clearer. The second one would not necessarily force plugin creators to validate / change their code, though.